### PR TITLE
Bitstamp-fetchTransactionFees-update

### PR DESCRIPTION
**Pull request by rayBastard**
Method `fetchTransactionFees` and `parseTransactionFees` according to new endpoint exchange

### CODE REVIEW

**Comment by rayBastard** [comment]
@carlosmiei I've updated `fetchDepositWithdrawFees` as well

**Comment by carlosmiei** [comment]
@rayBastard Hum, maybe we need to sanitize the currency ID, BTC is the unified currency not btc
```
n bitstamp fetchDepositWithdrawFees BTC          
2024-03-29T00:22:42.962Z
Node.js: v18.18.0
CCXT v4.2.84
bitstamp.fetchDepositWithdrawFees (BTC)
2024-03-29T00:22:43.391Z iteration 0 passed in 189 ms

{}
2024-03-29T00:22:43.391Z iteration 1 passed in 189 m
```

File: ts/src/bitstamp.ts [hunk 1]
```diff
@@ -1261,66 +1261,47 @@ export default class bitstamp extends Exchange {
...
-        let mainCurrencyId = undefined;
-        const ids = Object.keys (response);
+        const currencies = this.indexBy (response, 'currency');
+        const ids = Object.keys (currencies);
         for (let i = 0; i < ids.length; i++) {
             const id = ids[i];
-            const currencyId = id.split ('_')[0];
-            const code = this.safeCurrencyCode (currencyId);
-            if (codes !== undefined && !this.inArray (code, codes)) {
+            const fees = this.safeValue (response, i, {});
```
Comment (carlosmiei) [inline ts/src/bitstamp.ts:1290]: also, no more `safeValue` usage pls

**Comment by carlosmiei** [comment]
@rayBastard even after fixing the currencyId, it just returns 1 chain but we have many of them, please check other implementations like `bingx` for instance
```
n bitstamp fetchDepositWithdrawFees '["ETH"]' 
2024-03-29T00:26:50.880Z
Node.js: v18.18.0
CCXT v4.2.84
bitstamp.fetchDepositWithdrawFees (ETH)
2024-03-29T00:26:51.325Z iteration 0 passed in 216 ms

{
  ETH: {
    info: { currency: 'ens', fee: '1.00000000', network: 'ethereum' },
    withdraw: { fee: 1 },
    deposit: {},
    networks: 'ethereum'
  }
}
2024-03-29T00:26:51.325Z iteration 1 passed in 216 ms
```

File: ts/src/bitstamp.ts [hunk 2]
```diff
@@ -1261,66 +1261,47 @@ export default class bitstamp extends Exchange {
...
-        if (codes === undefined) {
-            codes = Object.keys (this.currencies);
-        }
         const result = {};
-        let mainCurrencyId = undefined;
-        const ids = Object.keys (response);
+        const currencies = this.indexBy (response, 'currency');
+        const ids = Object.keys (currencies);
         for (let i = 0; i < ids.length; i++) {
             const id = ids[i];
```
Comment (carlosmiei) [inline ts/src/bitstamp.ts:1289]: we should use `safeCurrencyCode(id)` here to convert from the exchange-specific format to the unified one

Comment (rayBastard) [reply ts/src/bitstamp.ts:1289]: I see, I'll fix it

**Comment by rayBastard** [comment]
@carlosmiei I've fixed method. Could you look at the build. Don't get the reason why has it failed?

**Comment by carlosmiei** [comment]
DEMO
```
Node.js: v18.18.0
CCXT v4.2.88
bitstamp.fetchDepositWithdrawFees (BTC)
2024-04-03T11:10:34.625Z iteration 0 passed in 251 ms

{
  BTC: {
    info: [
      { currency: 'btc', fee: '0.00000000', network: 'xrpl' },
      { currency: 'btc', fee: '0.00050000', network: 'bitcoin' }
    ],
    withdraw: { fee: 0.0005 },
    deposit: {},
    networks: {
      XRP: { withdraw: { fee: 0 }, deposit: {} },
      BTC: { withdraw: { fee: 0.0005 }, deposit: {} }
    }
  }
}
```

**Pull request merged**